src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -53,81 +68,99 @@ private boolean existByStation(Station station) {
+	private Optional<Section> getRelatedUpStationSection(Station upStation) {
+		return sections.stream()
+				.filter(generatedSection -> upStation.equals(generatedSection.getUpStation()))
+				.findAny();
+	}
 
-    private void prependSection(Section section) {
-        final Station downStation = section.getDownStation();
+	private void prependSection(Section section) {
+		final Station downStation = section.getDownStation();
 
-        if (equalsFirstStation(downStation)) {
-            sections.add(section);
-            return;
-        }
+		if (equalsFirstStation(downStation)) {
+			sections.add(section);
+			return;
+		}
 
-        Section relatedSection = getRelatedDownStationSection(downStation);
+		Section relatedSection = getRelatedDownStationSection(downStation)
+				.orElseThrow(() -> new IllegalArgumentException(NOT_EXIST_STATION));
 
-        relatedSection.downStationUpdate(section.getUpStation(), section.getDistance());
-        sections.add(section);
-    }
+		relatedSection.downStationUpdate(section.getUpStation());
+		relatedSection.divideDistance(section.getDistance());
+		sections.add(section);
+	}
 
-    private boolean equalsFirstStation(Station downStation) {
-        List<Station> stations = getStations();
-        Station firstStation = stations.get(0);
+	private boolean equalsFirstStation(Station downStation) {
+		List<Station> stations = getStations();
+		Station firstStation = stations.get(0);
 
-        return firstStation.equals(downStation);
-    }
+		return firstStation.equals(downStation);
+	}
 
-    private Section getRelatedDownStationSection(Station downStation) {
-        return sections.stream()
-                .filter(generatedSection -> downStation.equals(generatedSection.getDownStation()))
-                .findAny()
-                .orElseThrow(() -> new IllegalArgumentException(NOT_EXIST_STATION));
-    }
+	private Optional<Section> getRelatedDownStationSection(Station downStation) {
+		return sections.stream()
+				.filter(generatedSection -> downStation.equals(generatedSection.getDownStation()))
+				.findFirst();
+	}
 
-    public Station getLastDownStation() {
-        int lastIndex = sections.size() - 1;
+	public int count() {
+		return sections.size();
+	}
 
-        return sections.get(lastIndex).getDownStation();
-    }
+	public void deleteSection(Station station) {
 
-    public List<Station> getStations() {
-        return sections.stream()
-                .sorted()
-                .map(Section::getStations)
-                .flatMap(Collection::stream)
-                .distinct()
-                .collect(Collectors.toList());
-    }
+		Optional<Section> relatedUpStationSection = getRelatedUpStationSection(station);
+		Optional<Section> relatedDownStationSection = getRelatedDownStationSection(station);
 
-    public void deleteLastSection() {
-        sections.remove(sections.size() - 1);
-    }
+		if (!relatedUpStationSection.isPresent() && !relatedDownStationSection.isPresent()) {
```

Comment:
```suggestion
		if (relatedUpStationSection.isEmpty() && relatedDownStationSection.isEmpty()) {
```
이렇게 작성 해보실 수도 있을 것 같아요 :)